src/evolutions/src/main/scala/play/api/db/slick/evolutions/internal/DBApiAdapter.scala

```diff
@@ -54,38 +51,71 @@ private[evolutions] object DBApiAdapter {
 
     lazy val url: String = withConnection(_.getMetaData.getURL())
 
-    def getConnection(): Connection = dbConfig.db.source.createConnection()
+    override def getConnection(): Connection = getConnection(autocommit = true)
 
-    def getConnection(autocommit: Boolean): Connection = {
-      val conn = getConnection()
-      conn.setAutoCommit(autocommit)
-      conn
+    override def getConnection(autocommit: Boolean): Connection = {
```

Comment:
This implementation is now consistent with what we have in Play and fixes a possible connection leak if it fails to `setAutoCommit`.

Reply:
Is this related in any way to #445  ?

Reply:
It is possible. But needs further investigation to confirm.